src/rdkafka_admin.c

```diff
@@ -2735,3 +2837,400 @@ rd_kafka_DescribeConfigs_result_resources (
+
+        for (i = 0 ; i < (int)topic_cnt ; i++) {
+                rd_kafkap_str_t ktopic;
+                int32_t partition_cnt;
+                int j;
+
+                /* topic */
+                rd_kafka_buf_read_str(reply, &ktopic);
+
+                /* #partitions */
+                rd_kafka_buf_read_i32(reply, &partition_cnt);
+
+                for (j = 0 ; j < (int)partition_cnt ; j++) {
+                        int32_t partition;
+                        rd_kafka_topic_partition_t *skel;
+
+                        /* partition */
+                        rd_kafka_buf_read_i32(reply, &partition);
+
+                        skel = rd_kafka_topic_partition_list_add(offsets,
+                                                                 ktopic.str,
+                                                                 partition);
+                        /* low_watermark */
+                        rd_kafka_buf_read_i64(reply, &skel->offset);
+
+                        /* error_code */
+                        rd_kafka_buf_read_i16(reply, &skel->err);
+                }
+        }
+
+        if (--state->outstanding == 0) {
+                rd_kafka_op_t *rko_result = rd_kafka_admin_result_new(rko_req);
+                rd_list_init(&rko_result->rko_u.admin_result.results, 1,
+                             rd_kafka_topic_partition_list_free);
+                rd_list_add(&rko_result->rko_u.admin_result.results,
+                            rd_kafka_topic_partition_list_copy(offsets));
+                *rko_resultp = rko_result;
+                rd_kafka_AdminFanoutState_free(state);
+                return RD_KAFKA_RESP_ERR_NO_ERROR;
+        } else {
+                /* Still waiting for outstanding requests to return */
+                return RD_KAFKA_RESP_ERR__PARTIAL;
+        }
+
+err_parse:
+        if (rko_result)
+                rd_kafka_op_destroy(rko_result);
+
+        rd_snprintf(errstr, errstr_size,
+                    "DeleteRecords response protocol parse failure: %s",
+                    rd_kafka_err2str(err));
+
+        return err;
+}
+
+rd_kafka_resp_err_t rd_kafka_DeleteRecords (rd_kafka_t *rk,
+                                            rd_kafka_topic_partition_list_t *offsets,
+                                            const rd_kafka_AdminOptions_t *options,
+                                            rd_kafka_queue_t *rkqu,
+                                            int timeout_ms) {
+        static const struct rd_kafka_admin_worker_cbs cbs = {
+                rd_kafka_DeleteRecordsRequest,
+                rd_kafka_DeleteRecordsResponse_parse,
+        };
+        rd_kafka_resp_err_t err;
+        int i;
+        rd_list_t leaders;
+        struct rd_kafka_partition_leader *leader;
+
+
+        rd_list_init(&leaders, offsets->cnt,
+                     (void *)rd_kafka_partition_leader_destroy);
+        err = rd_kafka_topic_partition_list_query_leaders(rk, offsets, &leaders,
```

Comment:
Admin APIs need to be completely asynchronous, so we can't do this blocking thing here.

Reply:
Let me add the previously mentioned leader_req() to master branch and you can use that instead.